---
pageClass: "rule-details"
sidebarDepth: 0
title: "math/no-static-infinity-calculations"
description: "disallow static calculations that go to infinity"
---

# math/no-static-infinity-calculations

> disallow static calculations that go to infinity

- ‚ùó <badge text="This rule has not been released yet." vertical="middle" type="error"> **_This rule has not been released yet._** </badge>
- ‚öôÔ∏è This rule is included in `"plugin:math/recommended"`.
- üí° Some problems reported by this rule are manually fixable by editor [suggestions](https://eslint.org/docs/developer-guide/working-with-rules#providing-suggestions).

## üìñ Rule Details

This rule disallow static calculations that go to infinity.

<eslint-code-block>

<!-- eslint-skip -->

```js
/* eslint math/no-static-infinity-calculations: 'error' */

/* ‚úì GOOD */
x = Infinity;
x = -Infinity;
x = Number.POSITIVE_INFINITY;
x = Number.NEGATIVE_INFINITY;
x = 2 ** 1023 - 2 ** 971 + 2 ** 1023;

/* ‚úó BAD */
x = 2 ** 1024;
x = 2 ** 1024 - 2 ** 971;
x = 2 ** 1023 + 2 ** 1023;
x = 2 ** 1023 - 2 ** 970 + 2 ** 1023;
```

</eslint-code-block>

## üîß Options

Nothing.

## üìö Further reading

- [MDN - Infinity](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Infinity)
- [MDN - Number.POSITIVE_INFINITY](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/POSITIVE_INFINITY)
- [MDN - Number.NEGATIVE_INFINITY](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/NEGATIVE_INFINITY)

## üë´ Related rules

- [math/no-static-nan-calculations]

[math/no-static-nan-calculations]: ./no-static-nan-calculations.md

## üîç Implementation

- [Rule source](https://github.com/ota-meshi/eslint-plugin-math/blob/main/src/rules/no-static-infinity-calculations.ts)
- [Test source](https://github.com/ota-meshi/eslint-plugin-math/blob/main/tests/src/rules/no-static-infinity-calculations.ts)
- [Test fixture sources](https://github.com/ota-meshi/eslint-plugin-math/tree/main/tests/fixtures/rules/no-static-infinity-calculations)
