---
pageClass: "rule-details"
sidebarDepth: 0
title: "math/prefer-number-epsilon"
description: "enforce the use of Number.EPSILON instead of other ways"
---

# math/prefer-number-epsilon

> enforce the use of Number.EPSILON instead of other ways

- â— <badge text="This rule has not been released yet." vertical="middle" type="error"> **_This rule has not been released yet._** </badge>
- âš™ï¸ This rule is included in `"plugin:math/recommended"`.
- ğŸ”§ The `--fix` option on the [command line](https://eslint.org/docs/user-guide/command-line-interface#fixing-problems) can automatically fix some of the problems reported by this rule.

## ğŸ“– Rule Details

This rule aims to enforce the use of `Number.EPSILON` instead of other ways.

<eslint-code-block fix>

<!-- eslint-skip -->

```js
/* eslint math/prefer-number-epsilon: 'error' */

/* âœ“ GOOD */
x = Number.EPSILON;

/* âœ— BAD */
x = 2 ** -52;
x = Math.pow(2, -52);
x = 2.220446049250313e-16;
```

</eslint-code-block>

## ğŸ”§ Options

Nothing.

## ğŸ“š Further reading

- [MDN - Number.EPSILON](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/EPSILON)

## ğŸ” Implementation

- [Rule source](https://github.com/ota-meshi/eslint-plugin-math/blob/main/src/rules/prefer-number-epsilon.ts)
- [Test source](https://github.com/ota-meshi/eslint-plugin-math/blob/main/tests/src/rules/prefer-number-epsilon.ts)
- [Test fixture sources](https://github.com/ota-meshi/eslint-plugin-math/tree/main/tests/fixtures/rules/prefer-number-epsilon)
